### PR TITLE
Influx module user fixes

### PR DESCRIPTION
**Pull request by eugene-davis**
### What does this PR do?
Removes a collision between the client_args parameter and the inputs to user creation and updating in the salt module and state for Influxdb. This allows user management on a system require authentication.

Also updates the documentation for the grants parameter to make its usage more clear.

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/37670

### Tests written?

No

### Commits signed with GPG?

No

### CODE REVIEW

**Comment by rallytime** [comment]
@babilen or @hvnsweeting Would either of you like to take a look at this change and review?

File: salt/states/influxdb_user.py [hunk 1]
```diff
@@ -52,11 +53,12 @@ def present(name,
         example user present in influxdb:
           influxdb_user.present:
             - name: example
-            - password: somepassword
+            - passwd: somepassword
             - admin: False
-            - grants:
-                foo_db: read
+            - grants: {
```
Comment (babilen) [inline salt/states/influxdb_user.py]: Why not rely on yaml parsing for the dictionary here?

**Comment by babilen** [comment]
Change looks fine. Not entirely sure why the YAML example was changed though.

File: salt/states/influxdb_user.py (hunk 1 above)
Comment (eugene-davis) [reply salt/states/influxdb_user.py]: As written it didn't work, I'm not super familiar with the way YAML parsing works though.

**Comment by eugene-davis** [comment]
Ok I've dropped the documentation change

**Comment by eugene-davis** [comment]
@babilen Anything additional I should add or someone else I should flag to get this set to merge?

**Comment by babilen** [comment]
Looks great, thanks

**Comment by rallytime** [comment]
@eugene-davis Can you add a note about this change to the [Fluorine Release notes](https://github.com/saltstack/salt/blob/develop/doc/topics/releases/fluorine.rst)? We need to document this kind of an API change so people can update their states appropriately.

**Review by gtmanfred — APPROVED** [review]
I am ok with this, since there is no way to pass in password twice (once for password= and once for client_kwargs)

`influxdb.foo_function admin supersecret password=whatever`

And there isn't a good way for there to be a deprecation path, it would take at least 4 releases to get to the desired state. (2 to deprecated password -> passwd, and then two more to make client_password in kwargs -> password).

So I am ok with this as long as there is a big note in the release docs for fluorine.

**Comment by eugene-davis** [comment]
@rallytime @gtmanfred I've added to the release notes describing the change. Please let me know if this is sufficient.

**Review by gtmanfred — APPROVED** [review]

**Comment by eugene-davis** [comment]
@gtmanfred @rallytime Is there something else I need to do prior to merge?
I see some of the checks aren't passing but this doesn't seem to be unusual looking at past merged PRs.

**Comment by gtmanfred** [comment]
I was just waiting to see if any other core dev wanted to review it.

**Comment by gtmanfred** [comment]
Thanks for the PR!

**Pull request merged**